src/visualization/utils/colorMappingUtils.ts

```diff
@@ -168,9 +168,21 @@ export const generateSeriesToColorHex = (
       colors = DEFAULT_LINE_COLORS
     }
     const colorsHexArray = colors.map(value => value.hex)
-    const fillScale = getNominalColorScale(columnGroupMap, colorsHexArray)
+    const key = colorsHexArray.join('')
+    let fillScale
+    if (cache[key]) {
+      fillScale = cache[key]
+    } else {
+      /**
+       * this is an expensive operation to perform, so we want to cache the result and use the returned function
+       * Since the size of the columnGroupMap is the same in the context of each function, we can rely upon the
+       * colorsHexArray as our unique key identifier and rely upon that for caching the reuslts
+       */
+      fillScale = getNominalColorScale(columnGroupMap, colorsHexArray)
+      cache[key] = fillScale
+    }
     seriesToColorHex[id] = fillScale(colorIndex)
   })
 
-  return {...seriesToColorHex}
+  return seriesToColorHex
```

Comment:
Agreed 👍 

My only fear is that this was possibly for an edge case, I can't remember at the moment tho, so if we see weird behavior after we ship this, we might need to revisit this. 

i think this looks good.

Reply:
Is it possible the data this is generated from comes directly from an object in state? I've seen this pattern elsewhere when folks are trying to clone an object from state so that it's not frozen by immer anymore. If that's the reason for the clone, we might need it, and also need a `cloneDeep`.

Reply:
More specifically - I mean something that's in redux state. If we're cloning `seriestoColorHex` to manipulate it locally without being subject to the state freeze, a spread like `{...seriesToColorHex}` isn't going to cut it if there are nested objects or arrays.